spec/serializer_spec.rb

```diff
@@ -847,4 +847,24 @@ def serialize_primary(object, options = {})
       })
     end
   end
+
+  describe 'include validation' do
+    it 'raises an exception when join character is invalid' do
+      expect do
+        JSONAPI::Serializer.serialize(create(:post), include: 'long_comments');
+      end.to raise_error(JSONAPI::Serializer::InvalidIncludeError, "'long_comments' is not a valid include.  Did you mean 'long-comments' ?")
```

Comment:
Yup, totally fair ;)  Done!

One tool that I would recommend is `rubocop`.  It automatically checks for style, syntax, and code complexity metrics.  You can start off with a "todo" configuration file which will analyze your code and set a standard of where you code is actually at.  So if any code change makes something worse than it already is it will notify you.  You can also ignore rules globally or for different files, depending on what style you want to enforce.  It's a nice tool for double-checking PRs.  If you want I'd be happy to make a PR in this repo to show you.